src/operator/contrib/corner_pooling.cc

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*!
+ * Copyright (c) 2018 by Contributors
+ * \file corner_pooling.cc
+ * \brief corner pooling operator
+ * \author Jiajie Tang 
+*/
+
+#include "./corner_pooling-inl.h"
+
+#include "../elemwise_op_common.h"
+
+namespace mxnet {
+namespace op {
+
+void CornerPoolingParamParser(nnvm::NodeAttrs *attrs) {
+  CornerPoolingParam param;
+  param.Init(attrs->dict);
+  attrs->parsed = param;
+}
+
+
+static bool CornerPoolingType(const nnvm::NodeAttrs &attrs,
+                              std::vector<int> *in_attrs,
+                              std::vector<int> *out_attrs) {
+  out_attrs->at(0) = in_attrs->at(0);
```

Comment:
use ```TYPE_ASSIGN_CHECK```

Reply:
I have modified all the code you mentioned above. Please check my code, Thank you so much and Happy Christmas Eve!